cudax/include/cuda/experimental/__stream/get_stream.cuh

```diff
@@ -0,0 +1,79 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of CUDA Experimental in CUDA C++ Core Libraries,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _CUDAX__STREAM_GET_STREAM
+#define _CUDAX__STREAM_GET_STREAM
+
+#include <cuda/__cccl_config>
+
+#if defined(_CCCL_IMPLICIT_SYSTEM_HEADER_GCC)
+#  pragma GCC system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_CLANG)
+#  pragma clang system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_MSVC)
+#  pragma system_header
+#endif // no system header
+
+#include <cuda_runtime_api.h>
+
+#include <cuda/std/__concepts/__concept_macros.h>
+#include <cuda/std/__concepts/convertible_to.h>
+#include <cuda/std/__cuda/api_wrapper.h>
+#include <cuda/stream_ref>
+
+#include <cuda/experimental/__stream/stream.cuh>
+
+namespace cuda::experimental
+{
+namespace __get_stream
+{
+
+template <class _Tp>
+_LIBCUDACXX_CONCEPT __convertible_to_stream_ref = _CUDA_VSTD::convertible_to<_Tp, ::cuda::stream_ref>;
+
+template <class _Tp>
+_LIBCUDACXX_CONCEPT_FRAGMENT(
+  __has_member_get_stream_,
+  requires(const _Tp& __t)(requires(!__convertible_to_stream_ref<_Tp>),
+                           requires(_CUDA_VSTD::same_as<decltype(__t.get_stream()), ::cuda::stream_ref>)));
```

Comment:
I am wondering whether we want to relax that constraint and accept anything that is convertible to a `cuda::stream_ref` thoughts?

Reply:
I think its fine to keep them separate, it separates what is a stream and what has some reference to a stream

Reply:
I meant that a user might have a `::cudaStream_t get_stream() const` member function, that would not be accepted right now.

On the other hand we definitely want to reject a `cudax::stream get_stream() const` because that would leak the passed stream

Reply:
Do we want to reject qualified `cudax::stream`s? I user's type could have a member like:
```c++
auto get_stream() const -> const cudax::stream&;
```